posix/unix.c

```diff
@@ -373,7 +378,12 @@ int unix_accept4(unsigned socket, struct sockaddr *address, socklen_t *address_l
 			break;
 		}
 
-		if ((new = unixsock_alloc(&newid, s->type, nonblock)) == NULL) {
+		if (s->nonblock && (s->connect == NULL)) {
+			err = -EWOULDBLOCK;
+			break;
+		}
+
+		if ((new = unixsock_alloc(&newid, s->type, s->nonblock)) == NULL) {
```

Comment:
I think it would be best to create new unixsock with `refcnt = 2` and require `unixsock_put()` at the end.
Rationale - we have a reference to the new object, that might be removed from the tree by someone else meanwhile. If we handle and use this reference, it should be reflected in the refcnt (1 for presence in the tree + 1 for handling it by us after creation).

Reply:
Fair point. I've extended `unixsock_alloc` with an additional `refs` arg to pass `refs = 2` here.